### PR TITLE
IGNITE-12977 Documentation.

### PR DESCRIPTION
**Pull request by xtern**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: docs/_docs/restapi.adoc [hunk 1]
```diff
@@ -240,6 +243,69 @@ Similarly, the `get` command with `keyType=int` and `valueType=date` would be:
...
+    <entry key="uid"          value="java.util.UUID"/>
+    <entry key="name"         value="java.lang.String"/>
+    <entry key="organization" value="java.lang.Long"/>
+    <entry key="married"      value="java.lang.Boolean"/>
+    <entry key="salary"       value="java.lang.Float"/>
+   </map>
+  </property>
+ </bean>
+
+or Java class "Person" loaded on the server.
```
Comment (Silberfuchs1) [inline docs/_docs/restapi.adoc:292]: '...is loaded on the server.'

File: docs/_docs/restapi.adoc [hunk 2]
```diff
@@ -240,6 +243,69 @@ Similarly, the `get` command with `keyType=int` and `valueType=date` would be:
...
+
+For example, if there is no definition of the "Person" type on the server, the fields will be converted in accordance with the standard JSON types:
+[source, javascript]
+"uid": "7e51118b-eb15-4373-b57f-4984cb9cd7ac", // string
+"name": "John Doe",                            // string
+"organization": 5678901,                       // int
+"married": false,                              // boolean
+"salary": 156.1                                // double
+
+In case the `query entity` was set
```
Comment (Silberfuchs1) [inline docs/_docs/restapi.adoc:276]: If the `query entity` is set

**Review by Silberfuchs1 — COMMENTED** [review]
Some minor issues need to be corrected.

File: docs/_docs/restapi.adoc [hunk 3]
```diff
@@ -169,10 +169,13 @@ To set a custom expire time, set the system variable: `IGNITE_REST_SESSION_TIMEO
 ====
 
 == Data Types
-The REST API also supports Java built-in types for put/get operations via `keyType` and `valueType` optional parameters.
-Note that unless one of the below mentioned types are explicitly specified, the REST protocol exchanges the key-value data in `String` format.
+The REST API also supports <<Java built-in types>> and <<Custom user defined types>> for put/get operations via `keyType` and `valueType` optional parameters.
+
+Note that unless one of the mentioned below types are explicitly specified, the REST protocol exchanges the key-value data in `String` format.
```
[the remark this answers is not in the document]
Comment (xtern) [reply docs/_docs/restapi.adoc]: fixed, thanks

File: docs/_docs/restapi.adoc [hunk 4]
```diff
@@ -240,6 +243,69 @@ Similarly, the `get` command with `keyType=int` and `valueType=date` would be:
...
+  "salary": 156.1
+ }
+
+The following request puts this object into the cache `testCache` as a value with a type name `Person` and a key `1`.
+
+ http://[host]:[port]/ignite?cacheName=testCache&cmd=put&keyType=int&key=1&valueType=Person&val=%7B%0A+++++%22uid%22%3A+%227e51118b-eb15-4373-b57f-4984cb9cd7ac%22%2C%0A+++++%22name%22%3A+%22John+Doe%22%2C%0A+++++%22organization%22%3A+5678901%2C%0A+++++%22married%22%3A+false%2C%0A+++++%22salary%22%3A+156.1%0A++%7D&
+
+On the server side, the JSON value from this request will be converted to link:/docs/data-modeling/data-modeling#binary-object-format[binary object]. Field types are resolved in the following order:
+
+* If the type name is a `Java class` available on the server, the class field types with corresponding names will be used for JSON object field types resolving.
```
[the remark this answers is not in the document]
Comment (xtern) [reply docs/_docs/restapi.adoc]: fixed, thanks

File: docs/_docs/restapi.adoc [hunk 5]
```diff
@@ -240,6 +243,69 @@ Similarly, the `get` command with `keyType=int` and `valueType=date` would be:
...
+    <entry key="uid"          value="java.util.UUID"/>
+    <entry key="name"         value="java.lang.String"/>
+    <entry key="organization" value="java.lang.Long"/>
+    <entry key="married"      value="java.lang.Boolean"/>
+    <entry key="salary"       value="java.lang.Float"/>
+   </map>
+  </property>
+ </bean>
+
+or Java class "Person" is loaded on the server.
```
Comment (SomeFire) [inline docs/_docs/restapi.adoc:292]:
```suggestion
If Java class "Person" is loaded on the server, class fields will be filled with given values.
```

File: docs/_docs/restapi.adoc [hunk 6]
```diff
@@ -240,6 +243,69 @@ Similarly, the `get` command with `keyType=int` and `valueType=date` would be:
...
+
+For example, if there is no definition of the "Person" type on the server, the fields will be converted in accordance with the standard JSON types:
+[source, javascript]
+"uid": "7e51118b-eb15-4373-b57f-4984cb9cd7ac", // string
+"name": "John Doe",                            // string
+"organization": 5678901,                       // int
+"married": false,                              // boolean
+"salary": 156.1                                // double
+
+If the `query entity` is set
```
Comment (SomeFire) [inline docs/_docs/restapi.adoc:276]:
```suggestion
If the `query entity` is set, query fields will be filled with given values.
```

**Review by SomeFire — CHANGES_REQUESTED** [review]

**Pull request merged**